backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/diagrams/AbstractNodeMappingStyleProvider.java

```diff
@@ -79,6 +91,56 @@ private INodeStyle getNodeStyle(VariableManager variableManager, LabelStyleDescr
         return style;
     }
 
+    /**
+     * Returns <code>true</code> if the <em>nodeStyleDescription</em> represents a simple figure (square, ellipse or
+     * lozenge) and the {@link #nodeMapping} container represents his children in a list, <code>false</code> otherwise.
+     *
+     * @param variableManager
+     *            The variable manager
+     * @param nodeStyleDescription
+     *            the node style description that could be considered as a list item node style
+     * @return <code>true</code> if the <em>nodeStyleDescription</em> should be considered as a list item node style
+     */
+    private boolean shouldBeConsideredAsListItemNodeStyle(VariableManager variableManager, LabelStyleDescription nodeStyleDescription) {
+        if ((nodeStyleDescription instanceof SquareDescription || nodeStyleDescription instanceof EllipseNodeDescription || nodeStyleDescription instanceof LozengeNodeDescription)
+                && this.abstractNodeMapping.eContainer() instanceof ContainerMapping) {
+            ContainerMapping parentMapping = (ContainerMapping) this.abstractNodeMapping.eContainer();
+            LabelStyleDescription labelStyleDescription = new LabelStyleDescriptionProvider(this.interpreter, parentMapping).apply(variableManager);
+            return labelStyleDescription instanceof FlatContainerStyleDescription && ContainerLayout.LIST.equals(parentMapping.getChildrenPresentation());
+        }
+        return false;
+    }
+
+    private INodeStyle createListItemNodeStyle(VariableManager variableManager, LabelStyleDescription nodeStyleDescription) {
+        // @formatter:off
+        return ListItemNodeStyle.newListItemNodeStyle()
+                .backgroundColor("transparent") //$NON-NLS-1$
+                .build();
+        // @formatter:on
+    }
+
+    private INodeStyle createListNodeStyle(VariableManager variableManager, FlatContainerStyleDescription flatContainerStyleDescription) {
+        ColorDescriptionConverter backgroundColorProvider = new ColorDescriptionConverter(this.interpreter, variableManager.getVariables());
+        ColorDescriptionConverter borderColorProvider = new ColorDescriptionConverter(this.interpreter, variableManager.getVariables());
```

Comment:
No need to create two converters. They have the exact same configuration (and `variableManager.getVariables()` is relatively costly).